src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -1058,6 +1145,96 @@ impl<'a, 'tcx> PatternContext<'a, 'tcx> {
     }
 }
 
+fn search_for_adt_without_structural_match<'tcx>(tcx: TyCtxt<'tcx>,
+                                                 ty: Ty<'tcx>)
+                                                 -> Option<&'tcx AdtDef>
```

Comment:
I think this function could use a comment giving more context and perhaps pointing to some of the relevant issues. It's name is good but would be pretty opaque if you didn't understand what was going on.

Reply:
Good idea, added such a comment in b0b64dd